servicedirectory/endpoints_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package servicedirectory
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestEndpoint(t *testing.T) {
+	// <test setup code>
+	tc := testutil.SystemTest(t)
+	createNsBuf := new(bytes.Buffer)
+	createNsErr := createNamespace(createNsBuf, tc.ProjectID)
+	if createNsErr != nil {
+		fmt.Printf("Failed to create namespace in test setup. Skipping: %v", createNsErr)
```

Comment:
Is this a skip or failure situation? This is calling one of the sample functions (`createNamespace`). Same below.

(To skip a test in Go, call `t.Skip`.)

Reply:
In what situation should we fail vs skip?  a create namespace failure here shouldn't mean that the services_test fails necessarily so a skip might be the best idea

Reply:
In a way, this is testing the `createNamespace` sample. So, if that sample is failing, the test should fail.